anchor-tx/src/cas_remote.rs

```diff
@@ -0,0 +1,325 @@
+    pub alg: String,
+    pub kid: String,
+    pub typ: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Claims {
+    nonce: String,
+    url: String,
+    digest: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+struct AnchorResponse {
+    pub message: String,
+    pub witness_car: Option<String>,
+}
+
+fn cid_to_stream_id(cid: Cid) -> StreamId {
+    StreamId {
+        r#type: StreamIdType::Unloadable,
+        cid,
+    }
+}
+
+/// Remote CAS transaction manager
+pub struct RemoteCas {
+    signing_key: Ed25519KeyPair,
+    url: String,
+    poll_interval: Duration,
+    poll_retry_count: u32,
+    jws_header_b64: String,
+    http_client: reqwest::Client,
+    /// The did:key of the node represented as a StreamId
+    node_stream_id: StreamId,
+}
+
+enum CasResponseParseResult {
+    Anchored(Box<Receipt>),
+    Unauthorized,
+}
+
+#[async_trait]
+impl TransactionManager for RemoteCas {
+    async fn make_proof(&self, root: Cid) -> Result<Receipt> {
+        let mut interval = interval(self.poll_interval);
+        for _ in 0..self.poll_retry_count {
+            let anchor_response = self.create_anchor_request(root).await?;
+            match parse_anchor_response(anchor_response).await {
+                Ok(CasResponseParseResult::Anchored(receipt)) => {
+                    return Ok(*receipt);
+                }
+                Ok(CasResponseParseResult::Unauthorized) => {
+                    return Err(anyhow!("remote CAS request unauthorized"));
+                }
+                Err(e) => {
+                    debug!("swallowing anchoring result: {}", e);
+                }
+            }
+            interval.tick().await;
+        }
+        Err(anyhow::anyhow!(
+            "{} not anchored after {} attempts {}s apart",
+            root.to_string(),
+            self.poll_retry_count,
+            self.poll_interval.as_secs(),
+        ))
+    }
+}
+
+impl RemoteCas {
+    /// Create a new RemoteCas instance
+    pub fn new(
+        keypair: Ed25519KeyPair,
+        remote_anchor_service_url: String,
+        anchor_poll_interval: Duration,
+    ) -> Self {
+        let controller = did_key_from_ed25519_key_pair(&keypair);
+        let jws_header = Header {
+            kid: format!(
+                "{}#{}",
+                controller,
+                controller
+                    .strip_prefix("did:key:")
+                    .expect("invalid did:key")
+            ),
+            alg: "EdDSA".to_string(),
+            typ: "JWT".to_string(),
+        };
+        let jws_header_b64 =
+            b64.encode(serde_json::to_vec(&jws_header).expect("invalid jws header"));
+        let node_stream_id = cid_to_stream_id(cid_from_ed25519_key_pair(&keypair));
+        Self {
+            signing_key: keypair,
+            url: format!("{}/api/v0/anchor", remote_anchor_service_url),
+            poll_interval: anchor_poll_interval,
+            poll_retry_count: 12,
```

Comment:
if we keep this as a constant, can we make it a named constant at the top of the file?